src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -387,6 +387,10 @@ def _build_storage_profile():
     if availability_set_id:
         vm_properties['availabilitySet'] = {'id': availability_set_id}
 
+    # vmss is ID
+    if vmss is not None:
+        vm_properties['virtualMachineScaleSet'] = {'id': vmss}
```

Comment:
How are we supporting just the name? We need full ARM ID of VMSS when it hits the rest layer. So if you guys are converting that to ARM ID, this is fine. Or else follow what avset does.

Reply:
I handle this in _validate_vm_create_vmss(). This function can transform name to ID. I wrote examples in test cases.